frontend/public/components/row-filters.tsx

```diff
@@ -0,0 +1,348 @@
+import AutoCompleteInput from './auto-complete';
+import { Dropdown as DropdownInternal } from '@console/internal/components/utils';
+import { storagePrefix } from './row-filter';
+
+/**
+ * Housing both the row filter and name/label filter in the same file.
+ */
+
+enum FilterType {
+  NAME = 'Name',
+  LABEL = 'Label',
+}
+
+const filterTypeMap = Object.freeze({
+  [FilterType.LABEL]: 'labels',
+  [FilterType.NAME]: 'name',
+});
+
+type Filter = {
+  [key: string]: string[];
+};
+
+type FilterKeys = {
+  [key: string]: string;
+};
+
+const getDropdownItems = (rowFilters: RowFilter[], selectedItems, data) =>
+  rowFilters.map((grp) => (
+    <DropdownGroup
+      key={grp.filterGroupName}
+      label={grp.filterGroupName}
+      className="co-filter-dropdown-group"
+    >
+      {..._.map(grp.items, (item) => (
+        <DropdownItem
+          key={item.id}
+          id={item.id}
+          className="co-filter-dropdown__item"
+          listItemClassName="co-filter-dropdown__list-item"
+        >
+          <div className="co-filter-dropdown-item">
+            <span className="co-filter-dropdown-item__checkbox">
+              <Checkbox isChecked={selectedItems.includes(item.id)} id={`${item.id}-checkbox`} />
+            </span>
+            <span className="co-filter-dropdown-item__name">{item.title}</span>
+            <Badge key={item.id} isRead>
+              {_.countBy(data, grp.reducer)?.[item.id] ?? '0'}
+            </Badge>
+          </div>
+        </DropdownItem>
+      ))}
+    </DropdownGroup>
+  ));
+
+const getSelectedRowFilters = (rowFilters: RowFilter[]) =>
+  rowFilters.reduce((acc, curr) => [...acc, ...curr?.selected], []);
+
+const DropdownFilter_: React.FC<DropdownFilterProps> = (props) => {
+  const { rowFilters = [], data, showNameFilter = true, showLabelFilter = false } = props;
+
+  const [labelFilters, setLabelFilters] = React.useState<string[]>([]);
+  const [nameFilter, setNameFilter] = React.useState('');
+  const [labelFilterText, setLabelFilterText] = React.useState('');
+  const [filterType, setFilterType] = React.useState(FilterType.NAME);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [selected, setSelected] = React.useState(() => getSelectedRowFilters(rowFilters));
+
+  const filters: Filter = React.useMemo(
+    () =>
+      rowFilters.reduce((acc, curr) => {
+        const items = _.map(curr.items, 'id');
+        acc[curr.filterGroupName] = items;
+        return acc;
+      }, {}),
+    [rowFilters],
+  );
+
+  // { 'Xstatus' = 'x-y-status' ...}
+  const filterKeys: FilterKeys = React.useMemo(
+    () =>
+      rowFilters.reduce((acc, curr) => {
+        const str = `${storagePrefix}${curr.type}`;
+        acc[curr.filterGroupName] = str;
+        return acc;
+      }, {}),
+    [rowFilters],
+  );
+
+  // Parse state from URL
+  React.useEffect(() => {
+    let q: string = '',
+      name: string;
+    const rowFiltersFromURL: string[] = [];
+
+    if (window.location.search) {
```

Comment:
I would skip this check you have to test for missing values anyway. It simplies the logic and you can use `const`.

```ts
const params = new URLSearchParams(window.location.search);
const q = params.get('q') || '';
const name = params.get('name') || '';
// ...
```